code/modules/holodeck/computer.dm

```diff
@@ -169,12 +170,13 @@ GLOBAL_LIST_INIT(typecache_holodeck_linked_floorcheck_ok, typecacheof(list(/turf
 			if(program_to_load)
 				load_program(program_to_load)
 		if("safety")
-			if((obj_flags & EMAGGED) && program)
-				emergency_shutdown()
-			nerf(obj_flags & EMAGGED,FALSE)
-			obj_flags ^= EMAGGED
-			say("Safeties reset. Restarting...")
-			log_game("[key_name(usr)] disabled Holodeck safeties.")
+			if (obj_flags & EMAGGED | issilicon(usr))
+				if((obj_flags & EMAGGED) && program)
+					emergency_shutdown()
+				nerf(obj_flags & EMAGGED,FALSE)
+				obj_flags ^= EMAGGED
+				say("Safeties reset. Restarting...")
+				log_game("[key_name(usr)] disabled Holodeck safeties.")
```

Comment:
what exploit does this fix? obj_flags & EMAGGED | issilicon(usr) will always return true

Reply:
It seemed to work while testing, doesn't using emag on the console add the EMAGGED obj flag? I'm trying to make it so safety can't be changed unless it's either emagged or the user is a robot.

Reply:
Did you mean || ? Also sad to see this fixed, it’s the one of the oldest funny href exploits

Reply:
>Not reporting known href exploits

Bruh.

Reply:
obj_flags & EMAGGED was supposed to be (obj_flags & EMAGGED), but I'll play around with it and make it into a new pull once I stop being an idiot. Did anyone actually use this? I remember seeing people do the holodeck exploit before TGUI, it's not as easy to mess around with things anymore sadly. I find it funny that the exploit managed to make it's way over to TGUI though.